src/sentry/api/endpoints/group_notes.py

```diff
@@ -57,14 +59,30 @@ def post(self, request, group):
             reason=GroupSubscriptionReason.comment,
         )
 
-        if mentions:
-            users = User.objects.filter(id__in=mentions)
-            for user in users:
+        mentions = data.pop('mentions', [])
```

Comment:
did you intentionally move this? because now you're checking to see if the activity exists with a different data object than you're creating it with below